libbeat/processors/decode_xml/docs/decode_xml.asciidoc

```diff
@@ -0,0 +1,110 @@
+[[decode_xml]]
+=== Decode XML
+
+++++
+<titleabbrev>decode_xml</titleabbrev>
+++++
+
+experimental[]
+
+The `decode_xml` processor decodes any valid XML data that is stored under the `fields` key,
+It will output the values as an array of strings.
+
+Example that shows how to define which fields include the XML data and allows to overwrite the fields
+if they already exists:
+
+[source,yaml]
+-------
+processors:
+  - decode_xml:
+      fields: ["field1", "field2"]
+      overwrite_keys: true
+      target: ""
+-------
+
+If you want any potential errors to appear in the resulting document or want the automatically generated document ID to be overwritten by 
+ by a field that exists in the XML object an example like the below can be used:
+
+[source,yaml]
+-------
+processors:
+  - decode_xml:
+      fields: ["field1""]
+      add_error_key: true
+      target: "xml"
+      document_id: "fieldwithid"
+-------
+
+By default all keys converted from XML will have the names converted to lowercase. If there is a need to disable this behavior it is possible to use
+the below example:
+
+[source,yaml]
+-------
+processors:
+  - decode_xml:
+      fields: ["field1""]
+      to_lower: false
+      document_id: "fieldwithid"
+-------
+
+Example XML input:
+
+[source,xml]
+-------------------------------------------------------------------------------
+{
+  <catalog>
+    <book seq="1">
+      <author>William H. Gaddis</author>
+      <title>The Recognitions</title>
+      <review>One of the great seminal American novels of the 20th century.</review>
+    </book>
+  </catalog>
+}
+-------------------------------------------------------------------------------
+
+Will produce the following output:
+
+[source,json]
+-------------------------------------------------------------------------------
+{
+	"xml": {
+		"catalog": {
+			"book": {
+				"author": "William H. Gaddis",
+				"review": "One of the great seminal American novels of the 20th century.",
+				"seq": "1",
+				"title": "The Recognitions"
+			}
+		}
+	}
+}
+-------------------------------------------------------------------------------
+
+
+In the examples above:
```

Comment:
How about "The supported configuration options are:". This way it's clear that we're provided reference info.